src/CookieBanner.tsx

```diff
@@ -200,9 +200,11 @@ export default class CookieBanner extends React.Component<CookieBanner.Props, St
       <BannerContent {...bannerContentProps} />;
   }
 
-  componentWillReceiveProps(nextProps: CookieBanner.Props) {
-    if (nextProps.dismissOnScroll) {
-      this.addOnScrollListener(nextProps);
+  componentDidUpdate() {
+    const { props } = this;
+
+    if (props.dismissOnScroll) {
+      this.addOnScrollListener(props);
```

Comment:
LGTM. This is also what's been [recommended by Facebook](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#side-effects-on-props-change)